src/game/puppyprint.c

```diff
@@ -1226,6 +1228,62 @@ void get_char_from_byte(u8 letter, s32 *textX, s32 *textY, u8 *spaceX, s8 *offse
         case 'y': if (font == FONT_DEFAULT) *offsetY = 1; break;
     }
 }
+// This is where the deferred printing will be stored. When text is made, it will store text with an 12 byte header, then the rest will be the text data itself.
+// The first 4 bytes of the header will be the X and Y pos
+// The next 4 bytes will be the current envcolour set by print_set_envcolour
+// Then the string length, text alignment, amount and font each get a byte.
+// The data afterwards is the text data itself, using the string length byte to know when to stop.
+void print_small_text_buffered(s32 x, s32 y, const char *str, u8 align, s32 amount, u8 font) {
+    u8 strLen = MIN((signed)strlen(str), 255);
+    // Compare the cursor position and the string length, plus 11 (header size) and return if it overflows.
+    if (sPuppyprintTextBufferPos + strLen + 11 > sizeof(sPuppyprintTextBuffer))
+        return;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos] = (x >> 8) & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 1] = x & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 2] = (y >> 8) & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 3] = y & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 4] = gCurrEnvCol[0];
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 5] = gCurrEnvCol[1];
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 6] = gCurrEnvCol[2];
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 7] = gCurrEnvCol[3];
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 8] = strLen;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 9] = align;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 10] = (amount == -1) ? 255 : amount;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 11] = font;
+    bcopy(str, &sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 12], strLen);
+    sPuppyprintTextBufferPos += strLen + 12;
+}
+
+void puppyprint_print_deferred(void) {
+    if (sPuppyprintTextBufferPos == 0)
+        return;
+    print_set_envcolour(255, 255, 255, 255);
```

Comment:
Thinking for deferred printing, this should force an update to env color no matter what just in case. Maybe have this function return a boolean to see if the env color actually changed. If it returns FALSE, throw in a manual call to `gDPSetEnvColor(gDisplayListHead++, 255, 255, 255, 255);` instead.